### PR TITLE
service/dap: show decimal and hex values for uint

### PR DESCRIPTION
**Pull request by suzmue**
As discussed in golang/vscode-go#1459, users may want to see different representations of bytes and runes.

This change displays the hex value for all uints alongside the decimal value.
In addition to the variables response, this will impact the values in all evaluate contexts.

Examples:
```
myuint8: 255 = 0xff
myuint16: 65535 = 0xffff
myuint32: 4294967295 = 0xffffffff
myuint64: 18446744073709551615 = 0xffffffffffffffff
```

### CODE REVIEW

**Comment by suzmue** [comment]
@hyangah @polinasok

File: service/dap/server.go [hunk 1]
```diff
@@ -2107,6 +2108,9 @@ func (s *Server) convertVariableWithOpts(v *proc.Variable, qualifiedNameOrExpr s
 	}
 
 	switch v.Kind {
+	case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64, reflect.Uintptr:
+		n, _ := strconv.ParseUint(api.ConvertVar(v).Value, 10, 64)
+		value = fmt.Sprintf("%s = 0x%x", value, n)
```
Comment (aarzilli) [inline service/dap/server.go:2113]: `%#x`

**Review by aarzilli — COMMENTED** [review]
what happens if users that want to see unsigned integers as decimals exist (you wouldn't know if they did because they wouldn't have had a reason to speak until now)?

**Comment by suzmue** [comment]
> what happens if users that want to see unsigned integers as decimals exist (you wouldn't know if they did because they wouldn't have had a reason to speak until now)?

This change does not remove the decimal value, but instead displays both.

Screen Shot 2021-09-13 at 12 18 49 PM

 If we decide to display just one, we would definitely put this behind a configuration for these users.

File: service/dap/server.go (hunk 1 above)
Comment (suzmue) [reply service/dap/server.go]: Done.

**Review by aarzilli — APPROVED** [review]

**Comment by hyangah** [comment]
@suzmue Can you capture and share the screenshot of arrays? And this does not affect "Copy Value" and for Copy Value (clipboard context evaluate req) this will be still decimal, right?

**Comment by suzmue** [comment]
This is what it looks like in arrays:

Screen Shot 2021-09-14 at 3 42 38 PM

It currently applies to all contexts, but we could definitely change this to not include clipboard context.

**Comment by polinasok** [comment]
I suggest updating the PR description with the information about what will show up via Copy Value, hover, etc. 
And in case of uint8, we will have both the string child  and this feature?

**Review by polinasok — APPROVED** [review]

**Comment by suzmue** [comment]
Screen Shot 2021-09-15 at 2 27 26 PM

From offline discussion, we will for now leave both values in all contexts.

**Comment by polinasok** [comment]
> From offline discussion, we will for now leave both values in all contexts.

I updated the top PR comment.

**Pull request merged**